code-styling/markdown/README.md

```diff
@@ -0,0 +1,1155 @@
+  
+  ```Markdown
+  Please use `apt-get` to install packages on your Debian system.
+  ```
+  
+  _Incorrect example_:
+  
+  ```Markdown
+  Please use ` apt-get ` to install packages on your Debian system.
+  ```
+
+- Inline code containing backticks MAY use double backticks (``` `` ```) and in really rare cases the inline code
+  contains double backticks, triple backticks (```` ``` ````) MAY be used.
+
+  _Example_:
+  
+  ```Markdown
+  Displaying a single backtick (`` ` ``) is actually a very special and rare case.
+  ```
+
+- Shell code MUST NOT be prefixed with a dollar character (`$`).
+
+  _Correct example_:
+  
+  ```Markdown
+  Please use `apt-get` to install packages on your Debian system.
+  ```
+  
+  _Incorrect example_:
+  
+  ```Markdown
+  Please use `$ apt-get` to install packages on your Debian system.
+  ```
+
+- It is RECOMMENDED to use inline code for:
+
+  - executables
+  - file paths
+  - version numbers
+  - capitalized explanation of abbreviations
+  - specific computer and programming terms not found in the dictionary
+
+### 7.2. Code blocks
+
+- Code blocks MUST NOT be indent, unless the code block is placed inside a list.
+
+- Code blocks MUST be preceded and followed by a newline.
+
+- Code blocks MUST be fenced using three backtick characters (```` ``` ````).
+
+  _Correct example_:
+  
+  ```Markdown
+      To generate a PHP info page use this code:
+  
+      ```PHP
+      <?php
+        phpinfo();
+      ```
+  ```
+
+  _Incorrect example_:
+  
+  ```Markdown
+  To generate a PHP info page use this code:
+  
+    <?php
+      phpinfo();
+  ```
+
+- Every code block MUST have a language flag specified.
+
+- There MUST NOT be spacing between the backticks and the language flag.
+
+- When the code block is used to display monospaced text, `txt` MUST be used as the language flag.
+
+  _Correct example_:
+  
+  ```Markdown
+      ```txt
+      This is monospaced text.
+      ```
+  ```
+
+- Shell code MUST NOT be prefixed with a dollar character (`$`), unless you will be showing the command output on
+  the same code block.
+
+  _Correct example_:
+  
+  ```Markdown
+      ```txt
+      $ rubocop .
+      
+      Inspecting 16 files
+      ................
+    
+      16 files inspected, no offenses detected
+      ```
+  ```
+
+  _Incorrect example_:
+  
+  ```Markdown
+      ```txt
+      $ gem install rubocop
+      $ rubocop .
+      ```
+  ```
+
+- Code inside a code block MUST NOT be indented.
+
+_Note_: Code blocks shown inside the above code block examples are indented. Even tough this is against the rules in
+this guide, it is the only way of displaying a Markdown fenced code block inside a Markdown fenced code block.
+
+## 8. Links
+
+- The text that needs to be a link MUST be delimited by square brackets (`[`, `]`).
+
+- Links to internal pages MUST use relative paths.
+
+- Links MUST NOT have an extra trailing slash (`/`)
+
+  _Correct example_:
+  
+  ```Markdown
+  <http://www.example.com>
```

Comment:
See this rule about automatic linking:
- Links to websites SHOULD NOT use automatic links, unless the full link is displayed.

In this case, I want to show the difference between te URL linked, hence the use of the automatic link.